src/Mod/Sketcher/Gui/TaskSketcherTool.cpp

```diff
@@ -0,0 +1,967 @@
+    connect(ui->parameterFour, SIGNAL(valueChanged(double)),
+        this, SLOT(emitSetparameterFour(double)));
+    connect(ui->parameterFive, SIGNAL(valueChanged(double)),
+        this, SLOT(emitSetparameterFive(double)));
+    connect(ui->parameterSix, SIGNAL(valueChanged(double)),
+        this, SLOT(emitSetparameterSix(double)));
+    this->installEventFilter(this);
+    ui->parameterOne->installEventFilter(this);
+    ui->parameterTwo->installEventFilter(this);
+    ui->parameterThree->installEventFilter(this);
+    ui->parameterFour->installEventFilter(this);
+    ui->parameterFive->installEventFilter(this);
+    ui->parameterSix->installEventFilter(this);
+    ui->notice->setWordWrap(true);
+    isWidgetActive = 0;
+}
+
+SketcherToolWidget::~SketcherToolWidget(){}
+
+//pre-select the number of the spinbox when it gets the focus.
+bool SketcherToolWidget::eventFilter(QObject* object, QEvent* event)
+{
+    if (object == ui->parameterOne && event->type() == QEvent::FocusIn) {
+        ui->parameterOne->selectNumber();
+    }
+    else if (object == ui->parameterTwo && event->type() == QEvent::FocusIn) {
+        ui->parameterTwo->selectNumber();
+    }
+    else if (object == ui->parameterThree && event->type() == QEvent::FocusIn) {
+        ui->parameterThree->selectNumber();
+    }
+    else if (object == ui->parameterFour && event->type() == QEvent::FocusIn) {
+        ui->parameterFour->selectNumber();
+    }
+    else if (object == ui->parameterFive && event->type() == QEvent::FocusIn) {
+        ui->parameterFive->selectNumber();
+    }
+    if (event->type() == QEvent::KeyPress) {
+        /*If a key shortcut is required to work on sketcher when a tool using Tool Setting widget
+        is being used, then you have to add this key to the below section such that the spinbox
+        doesn't keep the keypress event for itself. Note if you want the event to be handled by 
+        the spinbox too, you can return false.*/
+        QKeyEvent* keyEvent = static_cast<QKeyEvent*>(event);
+        if (keyEvent->key() == Qt::Key_Control)
+        {
+            sketchView->keyPressed(1, SoKeyboardEvent::LEFT_CONTROL);
+            return true;
+        }
+        else if (keyEvent->key() == Qt::Key_Shift)
+        {
+            sketchView->keyPressed(1, SoKeyboardEvent::RIGHT_SHIFT);
+            return true;
+        }
+        else if (keyEvent->key() == Qt::Key_Alt)
+        {
+            sketchView->keyPressed(1, SoKeyboardEvent::RIGHT_ALT);
+            return true;
+        }
+        else if (keyEvent->key() == Qt::Key_Escape)
+        {
+            sketchView->keyPressed(1, SoKeyboardEvent::ESCAPE);
+            return true;
+        }
+        else if (keyEvent->key() == Qt::Key_Space)
+        {
+            sketchView->keyPressed(1, SoKeyboardEvent::SPACE);
+            return true;
+        }
+    }
+    return false;
+}
+
+void SketcherToolWidget::emitTogglePreserveFilletChamferCorner(bool val) {

```

Comment:
This is just some comment intended for the new framework.

The ToolDefaultWidget in the new framework is generic. It is not a problem if you need to add some checkboxes. However, these checkboxes need to be generic as the Parameters, something like checkboxOne, checkboxTwo.

If a tool needs a totally different widget, then a new widget should be get from the factory (that will eventually be implemented).